src/openloco/openloco.cpp

```diff
@@ -883,6 +895,239 @@ namespace openloco
+        }
+
+        if (is_editor_mode())
+        {
+            ebx = 0;
+        }
+
+        if (ebx2 < ebx)
+        {
+            ebx = ebx2;
+        }
+
+        game_command_nest_level--;
+        if (game_command_nest_level == 0)
+            return ebx;
+
+        if ((flagsBackup2 & 0x20) != 0)
+            return ebx;
+
+        call(0x0046DE2B);
+
+        if (ebx != 0 && _updating_company_id == _player_company[0])
+        {
+            _game_command_map_z = _game_command_map_z + 24;
+            call(0x0046DC9F);
+            _game_command_map_z = _game_command_map_z - 24;
+        }
+
+        return ebx;
+    }
+
+    static void sub_431908(company_id_t al, string_id dx)
+    {
+
+        registers regs;
+        regs.al = al;
+        regs.dx = dx;
+        call(0x431908, regs);
+    }
+
+    static uint32_t loc_4314EA()
+    {
+        game_command_nest_level--;
+        if (game_command_nest_level != 0)
+            return 0x80000000;
+
+        if (_updating_company_id != _player_company[0])
+            return 0x80000000;
+
+        if (_gameCommandFlags & 8)
+            return 0x80000000;
+
+        if (_9C68E6 != 0xFFFE)
+        {
+            windows::show_error(_9C68E8, _9C68E6);
+            return 0x80000000;
+        }
+
+        // advanced errors
+        if (_9C68D0 != (void*)-1)
+        {
+            auto tile = (tile_element*)_9C68D0;
+
+            switch (tile->type())
+            {
+                case element_type::unk_1: // 4
+                    _commonFormatArgs[0] = objectmgr::get<track_object>(tile->as_unk1()->track_object_id())->name;
+                    _commonFormatArgs[1] = companymgr::get(_errorCompanyId)->name;
+                    sub_431908(_errorCompanyId, string_ids::str_1421);
+                    return 0x80000000;
+
+                case element_type::unk_7: //0x1C
+                    _commonFormatArgs[0] = objectmgr::get<road_object>(tile->as_unk7()->road_object_id())->name;
+                    _commonFormatArgs[1] = companymgr::get(_errorCompanyId)->name;
+                    sub_431908(_errorCompanyId, string_ids::str_1421);
+                    return 0x80000000;
+
+                case element_type::station: // 8
+                {
+                    station* pStation = stationmgr::get(tile->as_station()->station_id());
+                    _commonFormatArgs[0] = pStation->name;
+                    _commonFormatArgs[1] = pStation->town;
+                    _commonFormatArgs[2] = companymgr::get(_errorCompanyId)->name;
+                    sub_431908(_errorCompanyId, string_ids::str_1421);
+                    return 0x80000000;
+                }
+
+                case element_type::unk_3: // 0x0C
+                {
+                    _commonFormatArgs[0] = companymgr::get(_errorCompanyId)->name;
+                    sub_431908(_errorCompanyId, string_ids::str_1422);
+                    return 0x80000000;
+                }
+
+                default:
```

Comment:
Is this needed if it just `break`s? There's a proper fallback at the end of the routine, anyway.

Reply:
https://stackoverflow.com/questions/5402745/gcc-switch-on-enum-retain-missing-warning-but-use-default